src/pybind/mgr/orchestrator/module.py

```diff
@@ -964,6 +964,18 @@ def _set_prometheus_access_info(self, username: Optional[str] = None, password:
         except ArgumentError as e:
             return HandleCommandResult(-errno.EINVAL, "", (str(e)))
 
+    @_cli_write_command('orch prometheus set-target')
+    def _set_prometheus_target(self, url: Optional[str] = None, inbuf: Optional[str] = None) -> HandleCommandResult:
```

Comment:
I think we might want to require the `url` arg to be passed and remove the `inbuf` arg. It doesn't seem like you're actually using inbuf and having url be optional makes `ceph orch prometheus set-target` a valid command which doesn't make sense (and will just end up adding `null` to the list of targets)

Reply:
Updated the PR